executor/adapter.go

```diff
@@ -1115,9 +1116,20 @@ func getEncodedPlan(sctx sessionctx.Context, p plannercore.Plan, genHint bool, n
 		sctx.GetSessionVars().StmtCtx.SetEncodedPlan(encodedPlan)
 	}
 	if genHint {
-		hints := plannercore.GenHintsFromPhysicalPlan(p)
-		hintStr = hint.RestoreOptimizerHints(hints)
-		sctx.GetSessionVars().StmtCtx.SetPlanHint(hintStr)
+		hints = plannercore.GenHintsFromPhysicalPlan(p)
+		sctx.GetSessionVars().StmtCtx.SetPlanHint(hints)
+	}
+
+	// remove duplicate hints
+	tableHints = make([]*ast.TableOptimizerHint, 0)
```

Comment:
Maybe we can set the length to `len(hints)` rather than `0`?

Reply:
done